ghc-lib-gen/src/Main.hs

```diff
@@ -148,25 +151,31 @@ calcParserModules = do
         , "-dep-makefile .parser-depends"
         , "-M"]
         ++ include_dirs
-        ++ ["-ignore-package ghc-lib-parser -package ghc", "-package base"]
-        ++ hs_source_dirs ++ ["-ighc-lib/stage0/libraries/ghc-heap/build"]
+        ++ ["-ignore-package ghc-lib-parser"
+           , "-ignore-package ghc"
+           -- , "-ignore-package ghci"
+           , "-package base"
+           ]
+        ++ hs_source_dirs
         ++ ["ghc-lib/stage0/compiler/build/Parser.hs"]
   putStrLn "# Generating 'ghc/.parser-depends'..."
   system_ cmd
 
   buf <- readFile' ".parser-depends"
+  -- The idea here is to turn lines like:
+  -- 'compiler/prelude/PrelRules.o : compiler/prelude/PrelRules.hs'
+  -- into just the module name, in this case 'PrelRules'.
   let depends = filter (not . isPrefixOf "#") (lines buf)
       depends' = filter (isSuffixOf ".hs") depends
-      srcPaths = map snd (mapMaybe (stripInfix ":") depends')
-      srcPaths' = map (replace "ghc-lib/stage0/" "") srcPaths
-      srcPaths'' = map (replace "ghc-lib/stage1/" "") srcPaths'
-      srcPaths''' = map snd (mapMaybe (stripInfix "/") srcPaths'')
-      srcPaths'''' = map snd (mapMaybe (stripInfix "/") srcPaths''')
-      srcPaths''''' = map (replace "build/GHC/" "GHC/") srcPaths''''
-      srcPaths'''''' = map (replace "/" ".") srcPaths'''''
-      srcs = map (dropSuffix ".hs") (filter (isSuffixOf ".hs") srcPaths'''''')
-      -- 'GHCi.FFI doesn't get deduced but is needed
-      -- 'HeaderInfo' because we prefer it here rather than `ghc-lib`
+      srcPaths = map (trim . snd) (mapMaybe (stripInfix ":") depends')
+      srcDirs = ghcLibParserHsSrcDirs lib
+      srcPaths' = map (replace "/" "") (foldl
```

Comment:
Shouldn’t we replace `/` by `.` instead to convert from filenames to module names? I would also suggest to do this at the same step where you drop the `.hs` suffix so that you have a single step that converts from filenames to module names.

Reply:
That was a bug :)